<launch>

  <arg name="port" value="/dev/ttyUSB0" />
  <arg name="name" value="hummingbird_1" />
  <arg name="quad_frame" value="/vicon/hummingbird_1/hummingbird_1" />
  <arg name="world_frame" value="/odom" />
  <arg name="loop" value="true" />
  <arg name="loop_point" value="2" />
  <arg name="file" value="trap_xy.txt" />
  <arg name="tuning_file" value="default_h.yaml" /> 

  <!-- Vicon connection node -->
  <include file="$(find vicon_bridge)/launch/vicon.launch" />

  <!-- Asctec TRPY Controller -->
  <node pkg="asctec_ll_updated" name="$(arg name)" type="asctec_ll_updated_node" output="screen" respawn="true">
     <param name="port" value="$(arg port)"/>
     <param name="ll_status_rate" value="1"/>
     <param name="imu_calc_data_rate" value="10"/>
     <param name="rc_data_rate" value="0"/>
     <param name="ctrl_out_rate" value="10"/>
     <param name="gps_data_advanced_rate" value="0"/>
  </node>

  <!-- Asctec Position Controller -->
  <node name="asctec_controller" 
        pkg="pc_asctec_sim" 
        type="controller_node"
	output="screen">

     <rosparam file="$(find pc_asctec_sim)/cfg/$(arg tuning_file)" command="load"/>

     <param name="name" value="$(arg name)"/>
     <param name="world_frame" value="$(arg world_frame)"/>
     <param name="quad_frame" value="$(arg quad_frame)"/>

  </node>

  <!-- Trajectory Generator -->
  <node name="waypoint_server" 
        pkg="pc_asctec_sim" 
        type="trajectory_file_node" 
        output="screen">

     <param name="name" value="$(arg name)"/>
     <param name="loop" value="$(arg loop)"/>
     <param name="loop_point" value="$(arg loop_point)"/>
     <param name="file" value="$(arg file)"/>
  </node>

</launch>
